plugins/node/opentelemetry-instrumentation-mysql2/test/mysql2-promise.test.ts

```diff
@@ -0,0 +1,980 @@
+  let pool: mysqlTypes.Pool;
+  let poolCluster: mysqlTypes.PoolCluster;
+  const provider = new BasicTracerProvider();
+  const testMysql = process.env.RUN_MYSQL_TESTS; // For CI: assumes local mysql db is already available
+  const testMysqlLocally = process.env.RUN_MYSQL_TESTS_LOCAL; // For local: spins up local mysql db via docker
+  const shouldTest = testMysql || testMysqlLocally; // Skips these tests if false (default)
+  const memoryExporter = new InMemorySpanExporter();
+
+  const getLastQueries = async (count: number) => {
+    const [rows] = await rootConnection.query<GeneralLogResult[]>({
+      sql: "SELECT * FROM mysql.general_log WHERE command_type = 'Query' ORDER BY event_time DESC LIMIT ? OFFSET 1",
+      values: [count],
+    });
+
+    return rows.map(row => {
+      if (typeof row.argument === 'string') {
+        return row.argument;
+      } else {
+        return row.argument.toString('utf-8');
+      }
+    });
+  };
+
+  before(async function () {
+    if (!shouldTest) {
+      // this.skip() workaround
+      // https://github.com/mochajs/mocha/issues/2683#issuecomment-375629901
+      this.test!.parent!.pending = true;
+      this.skip();
+    }
+    provider.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
+    rootConnection = await mysqlTypes.createConnection({
+      port,
+      user: 'root',
+      host,
+      password: rootPassword,
+      database,
+    });
+    if (testMysqlLocally) {
+      testUtils.startDocker('mysql');
+      // wait 15 seconds for docker container to start
+      this.timeout(20000);
+      await new Promise(resolve => setTimeout(resolve, 15000));
+    }
+  });
+
+  after(async function () {
+    await rootConnection.end();
+    if (testMysqlLocally) {
+      this.timeout(5000);
+      testUtils.cleanUpDocker('mysql');
+    }
+  });
+
+  beforeEach(async () => {
+    instrumentation.disable();
+    contextManager = new AsyncHooksContextManager().enable();
+    context.setGlobalContextManager(contextManager);
+    instrumentation.setTracerProvider(provider);
+    instrumentation.enable();
+    connection = await mysqlTypes.createConnection({
+      port,
+      user,
+      host,
+      password,
+      database,
+    });
+    pool = mysqlTypes.createPool({
+      port,
+      user,
+      host,
+      password,
+      database,
+    });
+    poolCluster = mysqlTypes.createPoolCluster();
+    // the implementation actually accepts ConnectionConfig as well,
+    // but the types do not reflect that
+    poolCluster.add('name', {
+      port,
+      user,
+      host,
+      password,
+      database,
+    });
+  });
+
+  afterEach(async () => {
+    context.disable();
+    memoryExporter.reset();
+    instrumentation.setConfig({});
+    instrumentation.disable();
+    await connection.end();
+    await pool.end();
+    if (isPoolClusterEndIgnoreCallback()) {
+      await poolCluster.end();
+    } else {
+      await poolCluster.end();
+    }
```

Comment:
I'm am guessing you could drop `function isPoolClusterEndIgnoreCallback()` below and simplify this block to:

```suggestion
    await poolCluster.end();
```

Reply:
Fixed in the following commit. In addition, the ESLint error is also fixed.
[ef5d5a6](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2525/commits/ef5d5a6d210aa7821c2ad9524d858840a90cf66b)